src/bridging/BridgingSdk/getQuoteWithBridging.ts

```diff
@@ -0,0 +1,194 @@
+    bridgeQuoteRequestWithoutAmount
+
+  // Estimate the expected amount of intermediate tokens received in CoW Protocol's swap
+  const swapParams: WithPartialTraderParams<TradeParameters> = {
+    ...rest,
+    chainId: sellTokenChainId,
+    sellToken: sellTokenAddress,
+    buyToken: intermediateToken,
+    buyTokenDecimals: intermediaryTokenDecimals,
+    amount: amount.toString(),
+    signer,
+  }
+  const { result: swapResult, orderBookApi } = await tradingSdk.getQuoteResults(swapParams, {
+    ...advancedSettings,
+    appData: {
+      metadata: {
+        hooks: mockedHook,
+      },
+    },
+  })
+  const intermediateTokenAmount = swapResult.amountsAndCosts.afterSlippage.buyAmount // Estimated, as it will likely have surplus
+
+  // Get the bridge result
+
+  const { bridgeResult, bridgeHook, appData } = await getBridgeResult({
+    quoteBridgeRequest,
+    swapResult,
+    bridgeQuoteRequestWithoutAmount,
+    provider,
+    intermediateTokenAmount,
+    signer,
+  })
+
+  // Update the receiver and appData (both were mocked before we had the bridge hook)
+  swapResult.tradeParameters.receiver = bridgeHook.recipient
+  const { fullAppData, appDataKeccak256 } = await generateAppDataFromDoc(appData)
+  swapResult.appDataInfo = {
+    fullAppData,
+    appDataKeccak256,
+    doc: appData,
+  }
+
+  // Return the quote results with the post swap order function
+  return {
+    swap: swapResult,
+    bridge: bridgeResult,
+    async postSwapOrderFromQuote() {
+      const quoteResults = {
+        result: {
+          ...swapResult,
+          signer,
+        },
+        orderBookApi,
+      }
+
+      return postSwapOrderFromQuote(quoteResults, { ...advancedSettings, appData })
+    },
+  }
+}
+
+/**
+ * Ge the base params (without the amount) for the bridge quote request
+ */
+async function getBaseBridgeQuoteRequest<T extends BridgeQuoteResult>(params: {
+  quoteBridgeRequest: QuoteBridgeRequest
+  provider: BridgeProvider<T>
+  getErc20Decimals: GetErc20Decimals
+}): Promise<QuoteBridgeRequestWithoutAmount> {
+  const { provider, getErc20Decimals, quoteBridgeRequest } = params
+  const { sellTokenChainId } = quoteBridgeRequest
+
+  const intermediateTokens = await provider.getIntermediateTokens(quoteBridgeRequest)
+
+  if (intermediateTokens.length === 0) {
+    throw new Error('No path found')
+  }
+
+  // We just pick the first intermediate token for now
+  const intermediateTokenAddress = intermediateTokens[0]
+
+  // Get intermediate token decimals
+  const intermediaryTokenDecimals = await getErc20Decimals(sellTokenChainId, intermediateTokenAddress)
+
+  // Get the gas limit estimation for the hook
+  const bridgeQuoteRequestWithoutAmount: QuoteBridgeRequestWithoutAmount = {
+    ...quoteBridgeRequest,
+    sellTokenAddress: intermediateTokenAddress,
+    sellTokenDecimals: intermediaryTokenDecimals,
+  }
+
+  return bridgeQuoteRequestWithoutAmount
```

Comment:
Nitpick:

```suggestion
  return {
    ...quoteBridgeRequest,
    sellTokenAddress: intermediateTokenAddress,
    sellTokenDecimals: intermediaryTokenDecimals,
  }
```

Reply:
The reason I do this intermediate step, is because I want to make sure the type is the right one, so I get compiler errors if I forget a prop, or doesn't match the return type in this case

in this case, i don't mind to use your approach since there's a return type in the function with the expected type